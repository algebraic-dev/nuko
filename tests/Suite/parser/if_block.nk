let a: Int = 
    if a
    then b
    else c