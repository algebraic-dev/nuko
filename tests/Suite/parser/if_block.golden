✓ That
 └Program
   ├typeDecls: Empty List
   ├letDecls:
   | └List
   |   └LetDecl
   |     ├declName: (Name: ValName a 0:4-0:5)
   |     ├declArgs: Empty List
   |     ├declBody:
   |     | └If
   |     |   ├Lower
   |     |   | ├Local:
   |     |   | | └Name: ValName a 1:7-1:8
   |     |   | └NoExt
   |     |   ├Lower
   |     |   | ├Local:
   |     |   | | └Name: ValName b 2:9-2:10
   |     |   | └NoExt
   |     |   ├Just
   |     |   | └Lower
   |     |   |   ├Local:
   |     |   |   | └Name: ValName c 3:9-3:10
   |     |   |   └NoExt
   |     |   └1:4-3:10
   |     ├declRet:
   |     | └TId
   |     |   ├Local:
   |     |   | └Name: TyName Int 0:7-0:10
   |     |   └NoExt
   |     └declExt: NoExt
   ├impDecls: Empty List
   └programExt: NoExt
